apollo-core/src/main/java/com/ctrip/framework/foundation/internals/provider/DefaultServerProvider.java

```diff
@@ -101,68 +107,95 @@ public String getProperty(String name, String defaultValue) {
   }
 
   private void initEnvType() {
+    /**
+     * support ‘apollo.env’ 
+     */
+    for (String env : ENVS) {
+      if(initEnvType(env)) {
+        return;
+      }
+    }
+    
+    // 4. Set environment to null.
+    m_env = null;
+    logger.info("Environment is set to null. Because it is not available in either (1) JVM system property 'env', (2) OS env variable 'ENV' nor (3) property 'env' from the properties InputStream.");
+  }
+
+  private boolean initEnvType(String env) {
     // 1. Try to get environment from JVM system property
-    m_env = System.getProperty("env");
+    m_env = System.getProperty(env);
     if (!Utils.isBlank(m_env)) {
       m_env = m_env.trim();
       logger.info("Environment is set to [{}] by JVM system property 'env'.", m_env);
-      return;
+      return true;
     }
-
+ 
     // 2. Try to get environment from OS environment variable
-    m_env = System.getenv("ENV");
+    m_env = System.getenv(StringUtils.normalizeSyctemEnv(env));
     if (!Utils.isBlank(m_env)) {
       m_env = m_env.trim();
       logger.info("Environment is set to [{}] by OS env variable 'ENV'.", m_env);
-      return;
+      return true;
     }
-
+ 
     // 3. Try to get environment from file "server.properties"
-    m_env = m_serverProperties.getProperty("env");
+    m_env = m_serverProperties.getProperty(env);
     if (!Utils.isBlank(m_env)) {
       m_env = m_env.trim();
       logger.info("Environment is set to [{}] by property 'env' in server.properties.", m_env);
-      return;
+      return true;
     }
-
-    // 4. Set environment to null.
-    m_env = null;
-    logger.info("Environment is set to null. Because it is not available in either (1) JVM system property 'env', (2) OS env variable 'ENV' nor (3) property 'env' from the properties InputStream.");
+    return false;
   }
-
+  
   private void initDataCenter() {
+    /**
+     * support ‘apollo.idc’ 
+     */
+    for (String idc : IDCS) {
```

Comment:
I think it's not necessary to support `apollo.idc` as apollo already supports `apollo.cluster`